### PR TITLE
Add fuzzer test for encoding/decoding of a grid with gain maps.

### PR DESCRIPTION
**Pull request by maryla-uc**
Move some utilities to aviftest_helpers.h/.cc

### CODE REVIEW

File: tests/gtest/avif_fuzztest_enc_dec_incr_experimental.cc [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      std::min(image->height, grid_rows * cell_height);
+
+  const avifImage* gain_map = image->gainMap.image;
+  if (gain_map != nullptr) {
+    std::vector<AvifImagePtr> gain_map_cells =
+        ImageToGrid(gain_map, grid_cols, grid_rows);
+    if (gain_map_cells.empty()) return;
+    ASSERT_EQ(gain_map_cells.size(), cells.size());
+    for (size_t i = 0; i < gain_map_cells.size(); ++i) {
+      cells[i]->gainMap.image = gain_map_cells[i].release();
```
Comment (y-guyon) [inline tests/gtest/avif_fuzztest_enc_dec_incr_experimental.cc:52]: What happens if the gain map has different dimensions than the image? Are the cells still valid?  
This is mostly by curiosity, for fuzzing it does not matter.

File: tests/gtest/aviftest_helpers.cc [hunk 2]
```diff
@@ -501,5 +501,85 @@ avifIO* AvifIOCreateLimitedReader(avifIO* underlyingIO, uint64_t clamp) {
...
+      }
+    }
+  }
+  return true;
+}
+
+bool IsOpaque(const avifImage* image) {
+  return avifImageUsesU16(image) ? IsOpaque<uint16_t>(image)
+                                 : IsOpaque<uint8_t>(image);
+}
```
Comment (y-guyon) [inline tests/gtest/aviftest_helpers.cc:527]: `avifImageIsOpaque()` can be used instead

**Review by y-guyon — APPROVED** [review]
Thanks for adding tests.

File: tests/gtest/aviftest_helpers.cc (hunk 2 above)
Comment (maryla-uc) [reply tests/gtest/aviftest_helpers.cc]: Fixed, thanks.

File: tests/gtest/avif_fuzztest_enc_dec_incr_experimental.cc (hunk 1 above)
Comment (y-guyon) [reply tests/gtest/avif_fuzztest_enc_dec_incr_experimental.cc:52]: Thanks for the explanation. Note that the current change does not test mismatched `grid_cols` or `grid_rows` and `colorGrid+gainMapNoGrid` or `colorNoGrid+gainMapGrid` but that is fine.

**Review by y-guyon — APPROVED** [review]

**Pull request merged**